tensornetwork/backends/jax/jitted_functions.py

```diff
@@ -149,3 +150,151 @@ def body_vector(i, vals):
     ]
 
   return jax_lanczos
+
+
+def _generate_arnoldi_factorization(jax):
+  """
+  Helper function to create a jitted arnoldi factorization. 
+  The function returns a function `_arnoldi_fact` which 
+  performs an m-step arnoldi factorization.
+
+  `_arnoldi_fact` computes an m-step arnoldi factorization 
+  of an input callable `matvec`, with m = min(`it`,`num_krylov_vecs`). 
+  `_arnoldi_fact` will do at most `num_krylov_vecs` steps. 
+  `_arnoldi_fact` returns arrays `kv` and `H` which satisfy 
+  the Arnoldi recurrence relation
+  ```
+  matrix @ Vm - Vm @ Hm - fm * em = 0
+  ```
+  with `matrix` the matrix representation of `matvec` and 
+  `Vm =  jax.numpy.transpose(kv[:it, :])`, 
+  `Hm = H[:it, :it]`, `fm = np.expand_dims(kv[it, :] * H[it, it - 1]`,1) 
+  and `em` a kartesian basis vector of shape `(1, kv.shape[1])` 
+  with `em[0, -1] == 1` and 0 elsewhere.
+
+  The arguments to `_arnoldi_fact` are:
+
+  Args:
+    matvec: The matrix vector product. This function has to be wrapped into 
+      `jax.tree_util.Partial`. `matvec` will be called as `matvec(x, *args)`
+      for an input vector `x`.
+    args: List of arguments to `matvec`.
+    v0: Initial state to `matvec`.
+    krylov_vectors: An array for storing the krylov vectors. The individual
+      vectors are stored as columns.a
+    H: Matrix of overlaps.
+    start: Integer denoting the start position where the first 
+      produced krylov_vector should be inserted into `krylov_vectors`
+    num_krylov_vecs: Number of krylov iterations, should be identical to 
+      `krylov_vectors.shape[0]`
+    eps: Convergence parameter. Iteration is terminated if the norm of a
+      krylov-vector falls below `eps`.
+
+  Returns:
+    kv: An array of krylov vectors
+    H: A matrix of overlaps 
+    it: The number of performed iterations.
+
+  """
+
+  @jax.jit
+  def modified_gram_schmidt_step_arnoldi(j: int, vals: List):
+    """
+    Single step of a modified gram-schmidt orthogonalization.
+    Args:
+      j: Integer value denoting the vector to be orthogonalized.
+      vals: A list of variables:
+        `vector`: The current vector to be orthogonalized
+        to all previous ones
+        `krylov_vectors`: jax.array of collected krylov vectors
+        `n`: integer denoting the column-position of the overlap 
+          <`krylov_vector`|`vector`> within `H`.
+    Returns:
+      updated vals.
+             
+    """
+    vector, krylov_vectors, n, H = vals
+    v = krylov_vectors[j, :]
+    h = jax.numpy.vdot(v, jax.numpy.ravel(vector))
+    H = jax.ops.index_update(H, jax.ops.index[j, n], h)
+    vector = vector - h * jax.numpy.reshape(v, vector.shape)
+    return [vector, krylov_vectors, n, H]
+
+  @partial(jax.jit, static_argnums=(5, 6, 7))
+  def _arnoldi_fact(matvec, args, v0, krylov_vectors, H, start, num_krylov_vecs,
```

Comment:
We should probably settle on what we are going to call the number of Krylov vectors in these various algorithms; in Lanczsos it looks like you call the same thing `ncv`. I personally like either that or `n_kry`; `num_krylov_vecs` feels really long to me but it's a matter of taste.

Reply:
You're right, we should be consistent here. 

I think long argument names are generally the preferable choice, in particular in backend modules that don't interface with the users.

Reply:
or let me rephrase: expressive argument names are preferable

Reply:
I suppose `num_krylov_vecs` is fine, then

Reply:
i usually use `ncv` for user-facing functions to keep it consistent with scipy. Is taht OK?